actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -89,15 +89,23 @@ def expanded_cache_key(key, view, template, digest_path)
       # If the partial is not already cached it will also be
       # written back to the underlying cache store.
       def fetch_or_cache_partial(cached_partials, template, order_by:)
-        order_by.index_with do |cache_key|
+        entries_to_write = {}
+
+        keyed_partials = order_by.index_with do |cache_key|
           if content = cached_partials[cache_key]
             build_rendered_template(content, template)
           else
-            yield.tap do |rendered_partial|
-              collection_cache.write(cache_key, rendered_partial.body)
-            end
+            rendered_partial = yield
+            entries_to_write[cache_key] = rendered_partial.body.to_str
```

Comment:
Is `body` absolutely definitely 150% guaranteed to be an OutputBuffer (or String) here, even given a particularly pathological template renderer etc?

(This is both "can we assume there's a `to_str`?", _and_ "are we allowed to force the value to a string?")

Reply:
Nah, it was a good question. It appears that `EmptyCollection#body` returns `nil` for instance.

However I do think we can enforce `nil` or `respond_to?(:to_str)`.